Source/Core/VideoCommon/VertexLoaderBase.cpp

```diff
@@ -96,45 +97,32 @@ class VertexLoaderTester : public VertexLoaderBase
   std::vector<u8> buffer_b;
 };
 
-template <class Function>
-static void GetComponentSizes(const TVtxDesc& vtx_desc, const VAT& vtx_attr, Function f)
+u32 VertexLoaderBase::GetVertexSize(const TVtxDesc& vtx_desc, const VAT& vtx_attr)
 {
-  if (vtx_desc.low.PosMatIdx)
-    f(1);
-  for (auto texmtxidx : vtx_desc.low.TexMatIdx)
-  {
-    if (texmtxidx)
-      f(1);
-  }
+  u32 size = 0;
+
+  size += Common::CountSetBits(vtx_desc.low.Hex & 0x1FF);
```

Comment:
Can you add a comment along the lines of "Each enabled TexMatIdx adds one byte, as does PosMatIdx"?